src/auto/buffered_input_stream.rs

```diff
@@ -0,0 +1,153 @@
+// This file was generated by gir (d8a605d) from gir-files (469db10)
+// DO NOT EDIT
+
+use Cancellable;
+use Error;
+use FilterInputStream;
+use InputStream;
+use ffi;
+use glib;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::boxed::Box as Box_;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct BufferedInputStream(Object<ffi::GBufferedInputStream, ffi::GBufferedInputStreamClass>): FilterInputStream, InputStream;
+
+    match fn {
+        get_type => || ffi::g_buffered_input_stream_get_type(),
+    }
+}
+
+impl BufferedInputStream {
+    pub fn new<P: IsA<InputStream>>(base_stream: &P) -> BufferedInputStream {
+        unsafe {
+            InputStream::from_glib_full(ffi::g_buffered_input_stream_new(base_stream.to_glib_none().0)).downcast_unchecked()
+        }
+    }
+
+    pub fn new_sized<P: IsA<InputStream>>(base_stream: &P, size: usize) -> BufferedInputStream {
+        unsafe {
+            InputStream::from_glib_full(ffi::g_buffered_input_stream_new_sized(base_stream.to_glib_none().0, size)).downcast_unchecked()
+        }
+    }
+}
+
+pub trait BufferedInputStreamExt {
+    fn fill<'a, P: Into<Option<&'a Cancellable>>>(&self, count: isize, cancellable: P) -> Result<isize, Error>;
+
+    fn fill_async<'a, P: Into<Option<&'a Cancellable>>, Q: Fn(Result<(), Error>) + Send + Sync + 'static>(&self, count: isize, io_priority: i32, cancellable: P, callback: Q);
+
+    fn get_available(&self) -> usize;
+
+    fn get_buffer_size(&self) -> usize;
+
+    fn peek(&self, buffer: &[u8], offset: usize) -> usize;
```

Comment:
The buffer needs to be mutable, the function reads into this buffer

Reply:
Seems this function need be manual :(